ReactAndroid/src/main/java/com/facebook/react/devsupport/DevServerHelper.java

```diff
@@ -218,8 +218,10 @@ public String getInspectorDeviceUrl() {
   /**
    * @return the host to use when connecting to the bundle server from the host itself.
    */
-  private static String getHostForJSProxy() {
-    return AndroidInfoHelpers.DEVICE_LOCALHOST;
+  private String getHostForJSProxy() {
+    String host = getDebugServerHost();
```

Comment:
`AndroidInfoHelpers.getServerHost()` cannot return null, as it is basically choosing amongst one of three hard-coded, non-null strings. And this method is not called directly by my "three-liner" anyway.
Also, even if I add a null-check in my method, a null host will blow up in the caller - `DevServerHelper.getJSBundleURLForRemoteDebugging()`, which relies on` getHostForJsProxy()` returning a non-null value.

TL;DR: Don't get me wrong - adding a null check is 30 seconds, but it will just not fix anything, as the whole code is written in a way where null for the host is not anticipated anywhere. But if you really insist - I'll commit a null check ASAP.

Long story: For these cases, I'm more of a fan of using proper JSR-308 `@Nullable` annotations to clearly indicate if a type includes null not. A defensive non-null check might save the current method invocation, but still blow up the call chain if the caller does not anticipate the returned value to be null - as in our case. With that said - while the ReactAndroid code does have `@Nullable` annotations here and there - it is not Java-8 compatible and does not have a JSR-308 null analyzer running as part of the build (right?). So you are also right that if `DevServerHelper.getDebugServerHost()` starts to return null for host, this change will go unnoticed at compile time.